sentry-rails/lib/sentry/rails/background_worker.rb

```diff
@@ -2,7 +2,7 @@ module Sentry
   class BackgroundWorker
     def _perform(&block)
       # make sure the background worker returns AR connection if it accidentally acquire one during serialization
-      ActiveRecord::Base.connection_pool.with_connection do
+      ::Rails.application.executor.wrap do
```

Comment:
After Rails 7.0, it'll capture the block's exception and report it, which will enter here again and cause infinite loop. So now I think this may be a bad idea 😕

Reply:
but our `@__sentry_captured` fix will avoid the loop right?

Reply:
Not, because the loop will be caused by the exception from SDK (if there's any), instead of the one we capture.
For example, if an event for `ZeroDivisionError` caused a `NoMethodError` during serialization, that `NoMethodError` will be captured and became a new event. If that `NoMethodError` is somehow a problem of the SDK code and/or problematic contextual data, it'll be an infinite loop.

Reply:
right but then we can just add another rescue here, SDK should fail gracefully internally.
Or move [this rescue](https://github.com/getsentry/sentry-ruby/blob/87b50d78fdfdc4e6946d2d71c84819cf6ed54386/sentry-ruby/lib/sentry/background_worker.rb#L55-L57) over to both `_perform` methods.

Reply:
Yeah but that feels ugly. And if this doesn't actually solve #1745, I don't see the necessity to implement it like this.

Reply:
ah ok just read the follow up in the other issue